inst/tests/tests.Rraw

```diff
@@ -2186,7 +2186,7 @@ if (ncol(DT)==2L) setnames(DT,c("A","B")) # else don't stop under torture with s
 test(714, DT[,z:=6:10], data.table(A=1:5,B=5,z=6:10))
 
 # Test J alias is now removed outside DT[...] from v1.8.7 (to resolve rJava::J conflict)
-test(715, J(a=1:3,b=4), error=base_messages$missing_function("J"))
+test(715, J(a=1:3,b=4), NULL)
```

Comment:
please just remove the argument instead of using explicit NULL

Reply:
It gives an error 
```
Test 715 ran without errors but failed check that x equals y:
> x = J(a = 1:3, b = 4) 
First 0 of 0 (type 'NULL'): 
NULL
> y = TRUE 
First 1 of 1 (type 'logical'): 
[1] TRUE
target is NULL, current is logical
```
I was trying to find the issue and got to know that y arguement of test defaults to true, since our test ouputs Null it causes the test to fail,

```
test = function(num,x,y=TRUE,error=NULL,warning=NULL,message=NULL,output=NULL,notOutput=NULL,ignore.warning=NULL,options=NULL,env=NULL) {
```

Reply:
I thought you were using NULL for the error arg, but actually it is the expected value (y arg).
so I guess that makes sense.
I wonder if it is desirable to keep that error though? What do you think @MichaelChirico @jangorecki  @ben-schwen ?

Reply:
Probably best to do something like with `:=` and instead of a stub returning `NULL`, make it a stub erroring "Hey you shouldn't call that here/there's probably a mistake":

https://github.com/Rdatatable/data.table/blob/e7b7e47280b34f1cb5323ffe9f78236a9307ff3c/R/data.table.R#L2768-L2771

Reply:
done :)

Reply:
ok that is better but the error could be improved. it should say in what context J should be used.
also do we want a similar error for `.` ?

Reply:
Yes i think having a similar error for `.` won't do any harm, anyways these function aren't supposed to called outside of data.table operations anyways.

Reply:
> ok that is better but the error could be improved. it should say in what context J should be used. also do we want a similar error for `.` ?

done :)